test/libvirt.go

```diff
@@ -12,14 +12,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package libvirt
+// Package test provides a mock libvirt server for RPC testing.
+package test
 
 import (
 	"encoding/binary"
 	"net"
 	"sync/atomic"
 )
 
+const (
```

Comment:
Let's put these in an internal package where they can be shared by both the base package and this testing package.